src/appleseed/renderer/kernel/rendering/generic/genericframerenderer.cpp

```diff
@@ -316,11 +323,17 @@ namespace
             {
                 set_current_thread_name("pass_manager");
 
+                const size_t start_pass = m_frame.get_initial_pass();
+
+                PermanentShadingResultFrameBufferFactory* buffer_factory =
```

Comment:
See my other comment in `frame.cpp`. We want to avoid those potentially dangerous casts.

Reply:
It will work without casting but the buffer factory needs to be permanent. Can I point that out in a comment and forget about the cast ?

Reply:
Why does it need to be permanent? What happens if it's not (and assuming you don't cast anymore)?

Reply:
If it's not permanent, it means it's ephemeral and it will create an empty tile. So there no point in saving this.

Reply:
I understand, but wouldn't be a correct behavior? If an ephemeral buffer is used, then you get empty checkpoints, no problem, no special case. It's kind of the same as saying, if there are no lights in the scene, the render will be black, so we should abort instead of rendering. We don't, there's no special case for a scene without any light.

Reply:
Agreed